src/psyclone/dynamo0p3.py

```diff
@@ -8474,14 +8464,16 @@ def __init__(self, call, parent_call, check=True):
             if dyn_argument.descriptor.stencil:
                 # Create a stencil object and store a reference to it in our
                 # new DynKernelArgument object.
-                stencil = DynStencil(dyn_argument.descriptor.stencil['type'])
+                stencil = LFRicArgStencil(
+                    dyn_argument.descriptor.stencil['type']
+                    )
                 dyn_argument.stencil = stencil
 
                 if dyn_argument.descriptor.stencil['extent']:
                     raise GenerationError("extent metadata not yet supported")
                     # if supported we would add the following
                     # line. However, note there is currently no setter
-                    # for extent in DynStencil so this would need to
+                    # for extent in LFRicArgStencil so this would need to
```

Comment:
I don't think this comment is true anymore as a dataclass allows its attributes to be set directly? Therefore, I think you can just delete the sentence beginning "However, ...".

Reply:
Indeed, no need for setters with a dataclass.

Reply:
Well, we could but in this case the code changes required are minimal so I wouldn't bother with implementing setters. (Unless of course we modify any of the members elsewhere too.)